### PR TITLE
(EAI-598) [UI] Investigate further code splitting

### PR DESCRIPTION
**Pull request by nlarew**
Jira: (EAI-598) [UI] Investigate further code splitting

## Changes

- Adds some additional code splitting & lazy loading to some components inside of the chat window view
- Note the build output includes more, smaller, chunks. There is still some significant heft to the Message component that I think comes from the LG Chat markdown. Need to look upstream for a fix there.

Before (current `main`):

```
...
build/es/ConversationProvider.js            67.30 kB │ gzip:  16.89 kB
build/es/lodash.js                         121.75 kB │ gzip:  30.95 kB
build/es/index3.js                         173.01 kB │ gzip:  35.20 kB
build/es/index7.js                         181.52 kB │ gzip:  54.19 kB
build/es/messageLinks.js                   210.30 kB │ gzip:  41.70 kB
build/es/index2.js                         313.03 kB │ gzip:  83.32 kB
build/es/ui-text.js                        328.27 kB │ gzip:  66.02 kB
build/es/ChatWindow.js                   2,060.51 kB │ gzip: 418.91 kB
✓ built in 5.17s
```

After (this PR):

```
...
build/es/ConversationProvider.js            67.33 kB │ gzip:  16.89 kB
build/es/ChatWindow.js                      75.68 kB │ gzip:  13.25 kB
build/es/ChatMessageFeed.js                 77.13 kB │ gzip:  16.42 kB
build/es/lodash.js                         120.77 kB │ gzip:  30.56 kB
build/es/index13.js                        131.94 kB │ gzip:  44.06 kB
build/es/index6.js                         160.88 kB │ gzip:  31.62 kB
build/es/index15.js                        202.81 kB │ gzip:  39.37 kB
build/es/InputBar.js                       305.09 kB │ gzip:  61.32 kB
build/es/index2.js                         313.00 kB │ gzip:  83.31 kB
build/es/MessageRating.js                  612.44 kB │ gzip: 116.19 kB
build/es/MessageContent2.js              1,201.38 kB │ gzip: 262.81 kB
✓ built in 5.02s
```

### CODE REVIEW

File: packages/mongodb-chatbot-ui/src/MessageRating.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    status,
+    errorMessage,
+    maxCommentCharacterCount,
+    messageRatingProps,
+  } = props;
+
+  const hasRating = messageRatingProps.value !== undefined;
+
+  // TODO: Use this to animate the transition after https://jira.mongodb.org/browse/LG-3965 is merged.
+  // const ratingCommentInputVisible = hasRating && status !== "submitted" && status !== "abandoned";
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/MessageRating.tsx:43]: should we have ticket for this?

File: packages/mongodb-chatbot-ui/src/MessageRating.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+              if (!characterCountExceeded) {
+                await submit(textarea?.value ?? "");
+              }
+            }}
+            textareaProps={{
+              onChange: (e) => {
+                const textarea = e.target as HTMLTextAreaElement;
+                setCharacterCount(textarea.value.length);
+              },
+              // @ts-expect-error Hacky fix for https://jira.mongodb.org/browse/LG-3964
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/MessageRating.tsx:85]: should we have separate ticket for this and the following changes that are dependent on these LG tickets?

**Review by mongodben — APPROVED** [review]
lgtm (mostly based on trust). what do you think is the actual performance impact of this? like how many MS imprvoement on load time?

**Comment by nlarew** [comment]
> what do you think is the actual performance impact of this

Honestly pretty minimal, though we'd need to test to get actual numbers. Should have some small effect though.

File: packages/mongodb-chatbot-ui/src/MessageRating.tsx (hunk 1 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/src/MessageRating.tsx:43]: Sure https://jira.mongodb.org/browse/EAI-618

**Pull request merged**